File: flut/flutter/material/button_style.py
from typing import Optional

from flut.dart.core import Duration
from flut.dart.ui import Color, Size
from flut._flut.engine import FlutRealtimeObject, _flut_pack_value
from flut.flutter.material.button_style_button import IconAlignment
from flut.flutter.material.ink_well import InteractiveInkFeatureFactory
from flut.flutter.material.theme_data import MaterialTapTargetSize, VisualDensity
from flut.flutter.painting.alignment import AlignmentGeometry
from flut.flutter.painting.borders import BorderSide, OutlinedBorder
from flut.flutter.painting.edge_insets import EdgeInsetsGeometry
from flut.flutter.painting.text_style import TextStyle
from flut.flutter.services.mouse_cursor import MouseCursor
from flut.flutter.widgets.widget_state import WidgetStateProperty


class ButtonStyle(FlutRealtimeObject):
    _flut_type = "ButtonStyle"

    def __init__(
        self,
        *,
        textStyle: Optional[WidgetStateProperty[TextStyle | None]] = None,
        backgroundColor: Optional[WidgetStateProperty[Color | None]] = None,
        foregroundColor: Optional[WidgetStateProperty[Color | None]] = None,
        overlayColor: Optional[WidgetStateProperty[Color | None]] = None,
        shadowColor: Optional[WidgetStateProperty[Color | None]] = None,
        surfaceTintColor: Optional[WidgetStateProperty[Color | None]] = None,
        elevation: Optional[WidgetStateProperty[float | None]] = None,
        padding: Optional[WidgetStateProperty[EdgeInsetsGeometry | None]] = None,
        minimumSize: Optional[WidgetStateProperty[Size | None]] = None,
        fixedSize: Optional[WidgetStateProperty[Size | None]] = None,
        maximumSize: Optional[WidgetStateProperty[Size | None]] = None,
        iconColor: Optional[WidgetStateProperty[Color | None]] = None,
        iconSize: Optional[WidgetStateProperty[float | None]] = None,
        iconAlignment: Optional[IconAlignment] = None,
        side: Optional[WidgetStateProperty[BorderSide | None]] = None,
        shape: Optional[WidgetStateProperty[OutlinedBorder | None]] = None,
        mouseCursor: Optional[WidgetStateProperty[MouseCursor | None]] = None,
        visualDensity: Optional[VisualDensity] = None,
        tapTargetSize: Optional[MaterialTapTargetSize] = None,
        animationDuration: Optional[Duration] = None,
        enableFeedback: Optional[bool] = None,
        alignment: Optional[AlignmentGeometry] = None,
        splashFactory: Optional[InteractiveInkFeatureFactory] = None,
    ) -> None:
        super().__init__()
        props = {}
        if textStyle is not None:
            props["textStyle"] = _flut_pack_value(textStyle)
        if backgroundColor is not None:
            props["backgroundColor"] = _flut_pack_value(backgroundColor)
        if foregroundColor is not None:
            props["foregroundColor"] = _flut_pack_value(foregroundColor)
        if overlayColor is not None:
            props["overlayColor"] = _flut_pack_value(overlayColor)
        if shadowColor is not None:
            props["shadowColor"] = _flut_pack_value(shadowColor)
        if surfaceTintColor is not None:
            props["surfaceTintColor"] = _flut_pack_value(surfaceTintColor)
        if elevation is not None:
            props["elevation"] = _flut_pack_value(elevation)
        if padding is not None:
            props["padding"] = _flut_pack_value(padding)
        if minimumSize is not None:
            props["minimumSize"] = _flut_pack_value(minimumSize)
        if fixedSize is not None:
            props["fixedSize"] = _flut_pack_value(fixedSize)
        if maximumSize is not None:
            props["maximumSize"] = _flut_pack_value(maximumSize)
        if iconColor is not None:
            props["iconColor"] = _flut_pack_value(iconColor)
        if iconSize is not None:
            props["iconSize"] = _flut_pack_value(iconSize)
        if iconAlignment is not None:
            props["iconAlignment"] = _flut_pack_value(iconAlignment)
        if side is not None:
            props["side"] = _flut_pack_value(side)
        if shape is not None:
            props["shape"] = _flut_pack_value(shape)
        if mouseCursor is not None:
            props["mouseCursor"] = _flut_pack_value(mouseCursor)
        if visualDensity is not None:
            props["visualDensity"] = _flut_pack_value(visualDensity)
        if tapTargetSize is not None:
            props["tapTargetSize"] = _flut_pack_value(tapTargetSize)
        if animationDuration is not None:
            props["animationDuration"] = _flut_pack_value(animationDuration)
        if enableFeedback is not None:
            props["enableFeedback"] = _flut_pack_value(enableFeedback)
        if alignment is not None:
            props["alignment"] = _flut_pack_value(alignment)
        if splashFactory is not None:
            props["splashFactory"] = _flut_pack_value(splashFactory)
        self._flut_create(props=props)

File: flut/.flutter/lib/flutter/material/button_style.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flut/error.dart';

class FlutButtonStyle with FlutRealtimeObject<ButtonStyle> {
  FlutButtonStyle.createFromData({
    required FlutRuntime runtime,
    required Map<String, dynamic> data,
    required ButtonStyle target,
  }) {
    initRealtimeFromData(runtime: runtime, data: data, target: target);
  }

  FlutButtonStyle.createFromObject({
    required FlutRuntime runtime,
    required int oid,
    required ButtonStyle target,
  }) {
    initRealtimeFromObject(
      runtime: runtime,
      oid: oid,
      type: 'ButtonStyle',
      target: target,
    );
  }

  static FlutButtonStyle flutCreate(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return FlutButtonStyle.createFromData(
      runtime: runtime,
      data: data,
      target: ButtonStyle(
        textStyle: runtime
            .unpackGenericField<WidgetStateProperty<TextStyle?>, TextStyle?>(
              data,
              'textStyle',
            ),
        backgroundColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'backgroundColor',
            ),
        foregroundColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'foregroundColor',
            ),
        overlayColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'overlayColor',
            ),
        shadowColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'shadowColor',
            ),
        surfaceTintColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'surfaceTintColor',
            ),
        elevation: runtime
            .unpackGenericField<WidgetStateProperty<double?>, double?>(
              data,
              'elevation',
            ),
        padding: runtime
            .unpackGenericField<
              WidgetStateProperty<EdgeInsetsGeometry?>,
              EdgeInsetsGeometry?
            >(data, 'padding'),
        minimumSize: runtime
            .unpackGenericField<WidgetStateProperty<Size?>, Size?>(
              data,
              'minimumSize',
            ),
        fixedSize: runtime
            .unpackGenericField<WidgetStateProperty<Size?>, Size?>(
              data,
              'fixedSize',
            ),
        maximumSize: runtime
            .unpackGenericField<WidgetStateProperty<Size?>, Size?>(
              data,
              'maximumSize',
            ),
        iconColor: runtime
            .unpackGenericField<WidgetStateProperty<Color?>, Color?>(
              data,
              'iconColor',
            ),
        iconSize: runtime
            .unpackGenericField<WidgetStateProperty<double?>, double?>(
              data,
              'iconSize',
            ),
        iconAlignment: runtime.unpackOptionalField<IconAlignment>(
          data,
          'iconAlignment',
        ),
        side: runtime
            .unpackGenericField<WidgetStateProperty<BorderSide?>, BorderSide?>(
              data,
              'side',
            ),
        shape: runtime
            .unpackGenericField<
              WidgetStateProperty<OutlinedBorder?>,
              OutlinedBorder?
            >(data, 'shape'),
        mouseCursor: runtime
            .unpackGenericField<
              WidgetStateProperty<MouseCursor?>,
              MouseCursor?
            >(data, 'mouseCursor'),
        visualDensity: runtime.unpackOptionalField<VisualDensity>(
          data,
          'visualDensity',
        ),
        tapTargetSize: runtime.unpackOptionalField<MaterialTapTargetSize>(
          data,
          'tapTargetSize',
        ),
        animationDuration: runtime.unpackOptionalField<Duration>(
          data,
          'animationDuration',
        ),
        enableFeedback: runtime.unpackOptionalField<bool>(
          data,
          'enableFeedback',
        ),
        alignment: runtime.unpackOptionalField<AlignmentGeometry>(
          data,
          'alignment',
        ),
      ),
    );
  }

  @override
  dynamic getRawProperty(String property) {
    switch (property) {
      case 'textStyle':
        return flutTarget.textStyle;
      case 'backgroundColor':
        return flutTarget.backgroundColor;
      case 'foregroundColor':
        return flutTarget.foregroundColor;
      case 'overlayColor':
        return flutTarget.overlayColor;
      case 'shadowColor':
        return flutTarget.shadowColor;
      case 'surfaceTintColor':
        return flutTarget.surfaceTintColor;
      case 'elevation':
        return flutTarget.elevation;
      case 'padding':
        return flutTarget.padding;
      case 'minimumSize':
        return flutTarget.minimumSize;
      case 'fixedSize':
        return flutTarget.fixedSize;
      case 'maximumSize':
        return flutTarget.maximumSize;
      case 'iconColor':
        return flutTarget.iconColor;
      case 'iconSize':
        return flutTarget.iconSize;
      case 'iconAlignment':
        return flutTarget.iconAlignment;
      case 'side':
        return flutTarget.side;
      case 'shape':
        return flutTarget.shape;
      case 'mouseCursor':
        return flutTarget.mouseCursor;
      case 'visualDensity':
        return flutTarget.visualDensity;
      case 'tapTargetSize':
        return flutTarget.tapTargetSize;
      case 'animationDuration':
        return flutTarget.animationDuration;
      case 'enableFeedback':
        return flutTarget.enableFeedback;
      case 'alignment':
        return flutTarget.alignment;
      case 'splashFactory':
        return flutTarget.splashFactory;
      case 'backgroundBuilder':
        return flutTarget.backgroundBuilder;
      case 'foregroundBuilder':
        return flutTarget.foregroundBuilder;
    }
    throw FlutUnknownPropertyException('ButtonStyle', property);
  }

  @override
  bool setProperty(String property, dynamic value) {
    throw FlutUnknownPropertyException('ButtonStyle', property);
  }

  @override
  dynamic callMethod(
    String method,
    List<dynamic> args,
    Map<String, dynamic> kwargs,
  ) {
    throw FlutUnknownMethodException(method);
  }
}
